src/MailCollector.php

```diff
@@ -170,8 +170,27 @@ public function post_getEmpty()
         $this->fields['is_active']    = 1;
     }
 
-    public function prepareInput(array $input, $mode = 'add'): array
+    public function prepareInput(array $input, $mode = 'add'): array|false
     {
+        $missing_fields = [];
+        if (($mode === 'add' || isset($input['mail_server'])) && empty($input['mail_server'])) {
```

Comment:
An input with `'mail_server' => null` will not be detected by `isset()`.
```suggestion
        if (($mode === 'add' || array_key_exists('mail_server', $input)) && empty($input['mail_server'])) {
```